Fix locale issue for numeric string conversions

jnysteen opened the pull request:
Using a generated .NET Core client from an application with a locale that formats numbers with a comma as a decimal separator will make the client send invalid numeric parameters to the API. Any numeric parameter with decimals will added to the URL with a comma ('0,15' instead of '0.15'), potentially breaking the API integration.
In my specific use case, the API interpreted '0,15' as '15', resulting in pretty funky errors.

This commit ensures that parameters are parsed with the invariant culture, fixing the mentioned error.

<!-- Enter details of the change here. Include additional tests that have been done, reference to the issue for tracking, etc. -->

<!-- Please check the completed items below -->
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [ ] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

jnysteen commented:
@mandrean (2017/08), @jimschubert (2017/09) ❤️ @frankyjuang (2019/09) @shibayan (2020/02)

wing328 commented:
I've updated the samples with your patch and have Appveyor CI tested the change but it reported the following issues:
```
To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
dotnet build samples\client\petstore\csharp-netcore\OpenAPIClient\Org.OpenAPITools.sln
Microsoft (R) Build Engine version 16.2.32702+c4012a063 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.
  Restore completed in 1.85 sec for C:\projects\openapi-generator\samples\client\petstore\csharp-netcore\OpenAPIClient\src\Org.OpenAPITools\Org.OpenAPITools.csproj.
  Restore completed in 3.44 sec for C:\projects\openapi-generator\samples\client\petstore\csharp-netcore\OpenAPIClient\src\Org.OpenAPITools.Test\Org.OpenAPITools.Test.csproj.
Client\ClientUtils.cs(105,42): error CS0103: The name 'CultureInfo' does not exist in the current context [C:\projects\openapi-generator\samples\client\petstore\csharp-netcore\OpenAPIClient\src\Org.OpenAPITools\Org.OpenAPITools.csproj]
Build FAILED.
Client\ClientUtils.cs(105,42): error CS0103: The name 'CultureInfo' does not exist in the current context [C:\projects\openapi-generator\samples\client\petstore\csharp-netcore\OpenAPIClient\src\Org.OpenAPITools\Org.OpenAPITools.csproj]
    0 Warning(s)
    1 Error(s)
Time Elapsed 00:00:08.64
Command exited with code 1
```
Ref: https://ci.appveyor.com/project/WilliamCheng/openapi-generator/builds/32716046

Can you please take a look when you've time?

jnysteen commented:
> I've updated the samples with your patch and have Appveyor CI tested the change but it reported the following issues:
> 
> ```
> To install additional .NET Core runtimes or SDKs:
>   https://aka.ms/dotnet-download
> dotnet build samples\client\petstore\csharp-netcore\OpenAPIClient\Org.OpenAPITools.sln
> Microsoft (R) Build Engine version 16.2.32702+c4012a063 for .NET Core
> Copyright (C) Microsoft Corporation. All rights reserved.
>   Restore completed in 1.85 sec for C:\projects\openapi-generator\samples\client\petstore\csharp-netcore\OpenAPIClient\src\Org.OpenAPITools\Org.OpenAPITools.csproj.
>   Restore completed in 3.44 sec for C:\projects\openapi-generator\samples\client\petstore\csharp-netcore\OpenAPIClient\src\Org.OpenAPITools.Test\Org.OpenAPITools.Test.csproj.
> Client\ClientUtils.cs(105,42): error CS0103: The name 'CultureInfo' does not exist in the current context [C:\projects\openapi-generator\samples\client\petstore\csharp-netcore\OpenAPIClient\src\Org.OpenAPITools\Org.OpenAPITools.csproj]
> Build FAILED.
> Client\ClientUtils.cs(105,42): error CS0103: The name 'CultureInfo' does not exist in the current context [C:\projects\openapi-generator\samples\client\petstore\csharp-netcore\OpenAPIClient\src\Org.OpenAPITools\Org.OpenAPITools.csproj]
>     0 Warning(s)
>     1 Error(s)
> Time Elapsed 00:00:08.64
> Command exited with code 1
> ```
> 
> Ref: https://ci.appveyor.com/project/WilliamCheng/openapi-generator/builds/32716046
> 
> Can you please take a look when you've time?

Fixed in: https://github.com/OpenAPITools/openapi-generator/pull/6194/commits/5c03d5a846890ee7c029af219c623518ae755eac

wing328 commented on modules/openapi-generator/src/main/resources/csharp-netcore/ClientUtils.mustache:
```diff
@@ -1,6 +1,7 @@
 {{>partial_header}}
 
 using System;
+using System.Globalization;
```
Can we add this line after `using System.Collections;` to follow the alphabetical order?

jnysteen replied:
Sure :) https://github.com/OpenAPITools/openapi-generator/pull/6194/commits/39dffb83590addc09fd63d04d0393e230ad97ba3

wing328 commented:
Updated samples and tested passed via https://ci.appveyor.com/project/WilliamCheng/openapi-generator/builds/32772568

Will update the samples via a separate commit after merging into master.
